### PR TITLE
Cyril/sort samples

### PR DESCRIPTION
**Pull request by cyrielo**
# Description
This PR enables the ability to sort sample based on sample name

## Type of change
- [x] New feature (non-breaking change which adds functionality)

## Impacted Areas in Application
List general components of the application that this PR will affect:

Which pages?
- [x] Sample Details Page

Specific components 

* Samples

# Testing Script:

### Login Page
* Successfully login to the application and get taken to the project page

### All Project Page
* Switch between projects 
* Search for a sample
* Navigate through different pages of samples
* Select Upload new sample button
* Select sample to see generated report

# Checklist:

- [x] I have run the testing script to make sure current functionality is unchanged
- [x] I have done relevant tests that prove my fix is effective or that my feature works
- [x] I have spent time testing out edge cases for my feature
- [x] I have updated the test script or pull request template if necessary
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/assets/javascripts/components/samples.jsx [hunk 1]
```diff
@@ -21,14 +22,33 @@ class Samples extends React.Component {
...
     };
+    this.sortCount = 0;
+  }
+
+  sortSamples() {
+    this.sortCount += 1;
+    let new_sort;
+    if(this.sortCount === 3) {
+      this.sortCount = 0;
+      new_sort = 'created_at,desc';
```
Comment (cdebourcy) [inline app/assets/javascripts/components/samples.jsx]: This means the 3rd time you click on the arrow, the samples are again sorted by date and not name. I think the user would not understand what's happening, given that the arrow is next to "Name". Is that what we want @eggerdesigns ?

Comment (eggerdesigns) [reply app/assets/javascripts/components/samples.jsx]: @cdebourcy when you click on it the third time I think that the arrow should disappear and the list is once again sorted by most recent upload. This isn't super intuitive but it gets the job done.

Comment (cdebourcy) [reply app/assets/javascripts/components/samples.jsx]: Makes sense. @cyrielo so please hide the arrow the 3rd time

**Comment by cdebourcy** [comment]
I like the new loading messages to indicate what's going on! But when it says "Sorting by date created", it doesn't actually do it, please fix that bug and I'll approve

File: app/assets/javascripts/components/samples.jsx [hunk 2]
```diff
@@ -21,14 +22,15 @@ class Samples extends React.Component {
       sampleIdsParams: this.fetchParams('ids') || [],
       allSamples: [],
       allProjects: [],
-      sort_query: currentSort.sort_query ? currentSort.sort_query  : `sort_by=${this.defaultSortBy}`,
+      sort_by: this.fetchParams('sort_by') || 'name,asc',
```
Comment (cdebourcy) [inline app/assets/javascripts/components/samples.jsx]: default should be newest to oldest
